js/src/reader/json.ts

```diff
@@ -0,0 +1,238 @@
+}
+
+function readFloatVector(field: any, column: any) {
+    switch (field.type.precision) {
+        // TODO: case "HALF":   return new Float16Vector(readNumeric(field, column, Uint16Array));
+        case 'SINGLE': return new Float32Vector(readNumeric(field, column, Float32Array));
+        case 'DOUBLE': return new Float64Vector(readNumeric(field, column, Float64Array));
+    }
+    throw new Error(`Unrecognized FloatingPoint { precision: ${field.type.precision} }`);
+}
+
+function readList(fieldObj: any, column: any) {
+    const field = fieldFromJSON(fieldObj);
+    const fieldNode = fieldNodeFromJSON(column);
+    const validity = readValidity(column);
+    const offsets = readData(Int32Array, column.OFFSET);
+    return { field, fieldNode, validity, offsets };
+}
+
+// "VALIDITY": [1,1],
+// "OFFSET": [0,7,14],
+// "DATA": ["49BC7D5B6C47D2","3F5FB6D9322026"]
+function readBinary(fieldObj: any, column: any) {
+    const field = fieldFromJSON(fieldObj);
+    const fieldNode = fieldNodeFromJSON(column);
+    const validity = readValidity(column);
+    const offsets = readData(Int32Array, column.OFFSET);
+    // There are definitely more efficient ways to do this... but it gets the
+    // job done.
+    const joined = column.DATA.join('');
+    let data = new Uint8Array(joined.length / 2);
+    for (let i = 0; i < joined.length; i += 2) {
+        data[i >> 1] = parseInt(joined.substr(i, 2), 16);
+    }
+    return { field, fieldNode, validity, offsets, data };
+}
+
+function readNumeric<T extends TypedArray>(fieldObj: any, column: any, ArrayConstructor: TypedArrayConstructor<T>) {
+    const field = fieldFromJSON(fieldObj);
+    const fieldNode = fieldNodeFromJSON(column);
+    const validity = readValidity(column);
+    const data = readData(ArrayConstructor, column.DATA);
+    return { field, fieldNode, validity, data };
+}
+
+function readInt64<T extends (Uint32Array|Int32Array)>(fieldObj: any, column: any, ArrayConstructor: TypedArrayConstructor<T>) {
+    const field = fieldFromJSON(fieldObj);
+    const fieldNode = fieldNodeFromJSON(column);
+    const validity = readValidity(column);
+    let data = new ArrayConstructor(column.DATA.length * 2);
+    for (let i = 0; i < column.DATA.length; ++i) {
+        data[2 * i  ] = column.DATA[i] >>> 0;
+        data[2 * i + 1] = Math.floor((column.DATA[i] / 0xFFFFFFFF));
+    }
+    return { field, fieldNode, validity, data };
+}
+
+function readData<T extends TypedArray>(ArrayConstructor: TypedArrayConstructor<T>, column: [number]) {
+    return new ArrayConstructor(column);
+}
+
+function readValidity(column: any) {
+    return readBoolean(column.VALIDITY);
+}
+
+function readBoolean(arr: Array<number>) {
+    let rtrn: Uint8Array = new Uint8Array(Math.ceil(arr.length / 8));
+    for (const {item, index} of arr.map((item: any, index: number) => ({item, index}))) {
+        rtrn[index / 8 | 0] |= item << (index % 8);
+    }
+    return rtrn;
```

Comment:
Should we use the [`BoolVector.pack`](https://github.com/apache/arrow/blob/master/js/src/vector/numeric.ts#L51) method here? I _think_ BoolVector's Uin8Array needs be padded with 0's till it aligns with the next 8'th byte